tests/NovelRT.Tests/Animation/SpriteAnimatorStateTest.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT License (MIT). See LICENCE.md in the repository root for more information.
+
+#include <gtest/gtest.h>
+#include <NovelRT.h>
+
+using namespace NovelRT;
+using namespace NovelRT::Maths;
+using namespace NovelRT::Animation;
+
+class SpriteAnimatorStateTest : public testing::Test {
+protected:
+  std::vector<std::shared_ptr<SpriteAnimatorState>> _states;
+
+  void SetUp() override {
+    _states = std::vector<std::shared_ptr<SpriteAnimatorState>>{
+      std::make_shared<SpriteAnimatorState>(),
+      std::make_shared<SpriteAnimatorState>(),
+      std::make_shared<SpriteAnimatorState>(),
+      std::make_shared<SpriteAnimatorState>(),
+      std::make_shared<SpriteAnimatorState>(),
+    };
+  }
+};
+
+void setUpRelationships(std::vector<std::shared_ptr<SpriteAnimatorState>> targets, std::vector<std::function<bool()>> conditions = std::vector<std::function<bool()>>()) {
+  for (size_t i = 0; i < targets.size(); i++) {
```

Comment:
Is this setting up a circular buffer of animation states?

Reply:
Its not a circular buffer, its a state machine.

Each state holds a collection of frames and an indefinite amount of references to other states, with functions evaluating to `bool` as the "transition" conditions.